models/yang/sonic/sonic-acl.yang

```diff
@@ -0,0 +1,221 @@
+module sonic-acl {
+	namespace "http://github.com/Azure/sonic-acl";
+	prefix sacl;
+	yang-version 1.1;
+
+	import ietf-inet-types {
+		prefix inet;
+	}
+
+	import sonic-common {
+		prefix cmn;
+	}
+
+	import sonic-port {
+		prefix prt;
+	}
+
+	import sonic-mirror-session {
+		prefix sms;
+	}
+
+	organization
+		"SONiC";
+
+	contact
+		"SONiC";
+
+	description
+		"SONIC ACL";
+
+	revision 2019-05-15 {
+		description
+			"Initial revision.";
+	}
+
+	container sonic-acl {
+
+		container ACL_TABLE {
+
+			list ACL_TABLE_LIST {
+				key "aclname";
+
+				leaf aclname {
+					type string {
+						pattern '[a-zA-Z0-9]{1}([-a-zA-Z0-9_]{0,71})';
+						length 1..72;
+					}
+				}
+
+				leaf policy_desc {
+					type string {
+						length 1..255 {
+							error-app-tag policy-desc-invalid-length;
+						}
+					}
+				}
+
+				leaf stage {
+					type enumeration {
+						enum INGRESS;
+						enum EGRESS;
+					}
+				}
+
+				leaf type {
+					type enumeration {
+						enum MIRROR;
+						enum L2;
+						enum L3;
+						enum L3V6;
+					}
+				}
+
+				leaf-list ports {
+					type leafref {
+						path "/prt:sonic-port/prt:PORT/prt:PORT_LIST/prt:ifname";
+					}
+				}
+			}
+		}
+
+		container ACL_RULE {
+
+			list ACL_RULE_LIST { 
+				key "aclname rulename";
+
+				leaf aclname {
+					type leafref {
+						path "../../../ACL_TABLE/ACL_TABLE_LIST/aclname";
+					}
+					must "(/cmn:operation/cmn:operation != 'DELETE') or " +
+						"count(current()/../../../ACL_TABLE/ACL_TABLE_LIST[aclname=current()]/ports) = 0" {
+							error-message "Ports are already bound to this rule.";
+						}
+				} 
+
+				leaf rulename {
+					type string;
```

Comment:
Key separator should not be used in key value. CVL separates multiple keys using key separator. The translation from JSON to YANG will fail and it can't even validate the constraint.

Reply:
As per ACL SME, acl table and acl rule name are never converted to uppercase, so they are case sensitive. So we can only set the max string length as 63 as a constraint here.

Reply:
**Resolved**